### PR TITLE
Avoid calling a stopped SparkContext

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

We shall avoid calling cancelJobGroup if the sc is already stopped.

We shall stop the engine to stop serving before we stop the sc during the shutdown hook

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @zhang1002 @ulysses-you @pan3793 @turboFei

**Comment by yaooqinn** [comment]
cc @hammertank

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 1]
```diff
@@ -168,12 +168,16 @@ private[kyuubi] object Utils extends Logging {
     System.getProperty(IS_TESTING.key) != null
   }
 
+  val DEFAULT_SHUTDOWN_PRIORITY = 100
+  val SERVER_SHUTDOWN_PRIORITY = 75
+  val SPARK_CONTEXT_SHUTDOWN_PRIORITY = 50
```
Comment (ulysses-you) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:173]: We can add some comment that this priority depends on `org.apache.spark.util.ShutdownHookManager.SPARK_CONTEXT_SHUTDOWN_PRIORITY`

**Review by ulysses-you — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:175]: a small or big one has a higher priority?

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:175]: big

**Review by pan3793 — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**